### PR TITLE
REF: remove iloc case from _convert_slice_indexer

### PR DESCRIPTION
**Pull request by jbrockmendel**
_convert_slice_indexer is turning out to be one of the stickier methods to figure out, xref #31658.

The case with kind=None is only called from one place in core.indexing, and that is not reached in the tests.  I'd like to either a) figure out a non-None kind to pass to it, or b) determine that it can never be reached so can be removed.  Any thoughts on this are very welcome.

### CODE REVIEW

**Comment by jreback** [comment]
sure

**Comment by jreback** [comment]
sure

File: pandas/core/indexes/numeric.py [hunk 1]
```diff
@@ -394,10 +394,9 @@ def _convert_scalar_indexer(self, key, kind: str):
 
     @Appender(Index._convert_slice_indexer.__doc__)
     def _convert_slice_indexer(self, key: slice, kind=None):
+        assert kind in ["loc", "getitem", None]
 
-        if kind == "iloc":
-            return super()._convert_slice_indexer(key, kind=kind)
-
+        # We always treat __getitem__ slicing as label-based
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/numeric.py]: I am not sure this comment is correct? (eg for a series with Int64Index, slicing is positional?)

Comment (jbrockmendel) [reply pandas/core/indexes/numeric.py]: this method is for Float64Index

Comment (jorisvandenbossche) [reply pandas/core/indexes/numeric.py]: Ah, missed that in the diff. Thanks!

**Pull request merged**